HHH-11147 - Allow enhanced entities to be returned in a completely uninitialized state

dreab8 opened the pull request:
the ```EntityType```  adds the value of ```unwrapProxy``` to the new ```SharesSessionContractImplementor#internalLoad``` so that it can be added to the ```LoadEventListener.LoadType type;``` (line 1168) and then be accessed and used by the ```DefaultLoadEventListener``` (line 297)

sebersole commented:
Do the tests pass for you with this?  I made some changes and tried to incorporate these changes into my work, but this test (FetchGraphTest) fails for me.  In a way I am not sure why it would not.

Part of the reason I did not integrate this directly is that I wanted to not track this value on `LoadEventListener.LoadType`.  There are some static references to that class that you are possibly altering there.  I changed that to use the LoadEvent instead.  

However, I run into failures.  I will push to my branch and maybe you could have a look?

dreab8 commented:
just finished running the tests and  apart the 28 failures of `org.hibernate.test.type.OffsetTimeTest` all passed.
I'm going to give a look at your work.

dreab8 commented:
I added 
`LoadEventListener.INTERNAL_LOAD_EAGER_UNWRAP_PROXY`
`LoadEventListener.INTERNAL_LOAD_LAZY_UNWRAP_PROXY
`LoadEventListener.INTERNAL_LOAD_NULLABLE_UNWRAP_PROXY`
this can be a solution to the issue with your the change to `LoadEvent` or probably as I added in the comment to the PR move the  `setShouldUnwrapProxy` before the `fireLoad( event, type );`
